src/middleware/trailing-slash/index.test.ts

```diff
@@ -0,0 +1,43 @@
+import { Hono } from '../../hono'
+import { trimTrailingSlash, appendTrailingSlash } from '.'
+
+describe('Resolve trailing slash', () => {
+  let app: Hono
+
+  it('Trim', async () => {
+    app = new Hono({ strict: true })
+
+    app.use('*', trimTrailingSlash())
+
+    app.get('/the/example/endpoint/without/trailing/slash', async (c) => {
+      return c.text('301 moved permanently')
+    })
+
+    const resp = await app.request('/the/example/endpoint/without/trailing/slash/')
+    expect(resp).not.toBeNull()
+    expect(resp.status).toBe(301)
+  })
+
+  it('Append', async () => {
+    app = new Hono({ strict: true })
+
+    app.use('*', appendTrailingSlash())
+
+    app.get('/the/example/endpoint/with/trailing/slash/', async (c) => {
+      return c.text('301 moved permanently')
+    })
+    app.get('/the/example/simulate/a.file', async (c) => {
+      return c.text('200 ok')
+    })
+
+    let resp: Response
+
+    resp = await app.request('/the/example/endpoint/with/trailing/slash')
+    expect(resp).not.toBeNull()
+    expect(resp.status).toBe(301)
+
+    resp = await app.request('/the/example/simulate/a.file')
+    expect(resp).not.toBeNull()
+    expect(resp.status).toBe(200)
+  })
```

Comment:
Can you add a test when the request path is `/the/example/endpoint/with/trailing/slash/`?

Reply:
I'd consider that as unneeded. Middlewares are designed for specific usages.

Unifying paths for their application is the developers' job, and adding a not matched test is probably not a good idea since redirecting all (or specified) paths is the expected behavior, but automatically detect the routes isn't.